extensions/vercast/src/index.tsx

```diff
@@ -102,3 +123,85 @@ function Main(): JSX.Element {
     </List>
   )
 }
+
+
+function UpdateEnvironmentVariable(props: { projectId: string, projectName: string }): JSX.Element {
+  // Get props values
+  const projectId = props.projectId;
+  const projectName = props.projectName;
+
+  // Get preference values
+  const token = String(preferences.token.value)
+  if (token.length !== 24) {
+    showToast(ToastStyle.Failure, 'Invalid token detected')
+    throw new Error('Invalid token length detected')
+  }
+
+  // Setup useState objects
+  const [environmentVariables, setEnvironmentVariables] = useState<EnvironmentVariable[]>()
+
+  // On form submit function
+  const { pop } = useNavigation();
+  async function handleSubmit(values: FormValues) {
+    if (!validateForm(values)) {
+      return;
+    }
+    const envConf = JSON.parse(values.env_conf)
+    const updatedEnvVariable = await updateEnvironmentVariable(projectId, envConf.envId, envConf.envKey, values.env_value)
+
+    if(!updatedEnvVariable || !updatedEnvVariable.key){
+      showToast(ToastStyle.Failure, "Couldn't update variable");
+    }else{
+      showToast(ToastStyle.Success, updatedEnvVariable.key+" was updated!");
+      pop();
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedEnvironmentVariables = await fetchEnvironmentVariables(
+        projectId
+      )     
+      setEnvironmentVariables(fetchedEnvironmentVariables)
+    }
+    fetchData()
+  }, [projectId])
+  return (
+    <Form 
+      navigationTitle={projectName+" ->  Environment Variables"}
+      isLoading={!environmentVariables} 
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section title="Environment Variable">
+            <SubmitFormAction 
+              title="Update Variable"
+              icon={Icon.Pencil}
+              onSubmit={handleSubmit}
+               />
+            </ActionPanel.Section>
+        </ActionPanel>
+      }>
+      <Form.Dropdown 
+        id="env_conf" 
+        title="Environment Variable">
+          {environmentVariables?.map((ev) => {
+            const randomID = randomUUID()
+            return (
+              <Form.Dropdown.Item 
+                value={JSON.stringify({envId : ev.id, envKey : ev.key})} 
+                title={ev.key} />
+            )
+          })}
+      </Form.Dropdown>
+      <Form.TextArea id="env_value" title="New Value" />
+    </Form>
+  )
+}
```

Comment:
Would you mind moving this to a new file called `actions.ts`?

Reply:
Done ✅